### PR TITLE
Rework ACM defaults

### PR DESCRIPTION
**Pull request by rhaschke**
This is a first implementation of #2936, considering ACM defaults as a fallback again (instead of an overrride).
This **depends on https://github.com/ros-planning/srdfdom/pull/97**, which implements corresponding SRDF extensions.

I encountered one difficulty implementing this, namely serializing and deserializing ACM messages: 
- Always the full matrix is serialized (there is no simple/compact alternative)
- The previous code didn't correctly serialize pair entries (only if a pair was specified, the value was serialized).
  Thus, when a default was specified, but no pairs, the wrong value was transmitted.
- Now, only if the pair deviates from the expected default, an explicit pair entry is created on the receiving side to keep the matrix as compact as possible.

An example application can be found in https://github.com/ros-planning/panda_moveit_config/pull/95.

### CODE REVIEW

**Comment by rhaschke** [comment]
Hm. For some reason, CI fails while it works locally. Need to debug CI...

**Comment by rhaschke** [comment]
Cleaning the ccache, resolved the issue: https://github.com/ros-planning/moveit/runs/4131508895?check_suite_focus=true

**Comment by v4hn** [comment]
> Always the full matrix is serialized (there is no simple/compact alternative)

https://github.com/ros-planning/moveit/issues/964 is quite related here.

**Comment by rhaschke** [comment]
@v4hn, @JafarAbdi: I finally managed to adapt to `srdfdom` API changes requested by Michael in https://github.com/ros-planning/srdfdom/pull/97#discussion_r766844841. Note that MSA still only supports disabling collision pairs. Allowing to define defaults there as well, is another refactoring that I would like to factor out into another PR.

**Review by JafarAbdi — APPROVED** [review]

File: moveit_core/collision_detection/include/moveit/collision_detection/collision_matrix.h [hunk 1]
```diff
@@ -138,52 +137,47 @@ class AllowedCollisionMatrix
...
+   *  With each of the the known names in the collision matrix, form a pair using the name
+   *  specified as argument to this function and set the entry as indicated by \e allowed.
    *  @param name the object name
    *  @param allowed If false, indicates that collisions between two elements must be checked for and no collisions
    *  will be ignored (AllowedCollision::NEVER). If true, indicates that collisions between two elements are ok and an
-   * explicit collision
-   *  computation is not necessary (AllowedCollision::ALWAYS).*/
-  void setEntry(const std::string& name, bool allowed);
+   *  explicit collision computation is not necessary (AllowedCollision::ALWAYS).*/
+  [[deprecated]] void setEntry(const std::string& name, bool allowed);
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply moveit_core/collision_detection/include/moveit/collision_detection/collision_matrix.h:161]: Good catch. I think this is a left-over of my intermediate check for usages of this function, which I indeed suggest to replace with `setDefaultEntry()`. However, I don't think we should deprecate the existing function.

**Comment by rhaschke** [comment]
@wolfv, RoboStack is failing here due to expected API changes in the upstream package `srdfdom` (https://github.com/ros-planning/srdfdom/pull/97). Do these changes automatically become active as soon as we release `srdfdom` or are there any actions required on your side?

**Comment by wolfv** [comment]
@rhaschke we will need to rebuild this package as part of the [robostack/ros-noetic](http://github.com/robostack/ros-noetic) build pipelines once it's released into noetic. Do you have an estimate when that will happen?
How do you deal with this in the other CI pipelines?

**Comment by rhaschke** [comment]
> Do you have an estimate when that will happen?

I plan to have a new release *after* the next sync (which is not yet scheduled).

> How do you deal with this in the other CI pipelines?

[`industrial_ci`](https://github.com/ros-industrial/industrial_ci) allows to specify source repositories as upstream:
https://github.com/ros-planning/moveit/blob/424a5b7b8b774424f78346d1e98bf1c9a33f0e78/.github/workflows/upstream.rosinstall#L9-L12

**Pull request merged**